receive.go

```diff
@@ -2,160 +2,78 @@ package main
 
 import (
 	"context"
-	"fmt"
 
+	"github.com/certusone/wormhole/node/pkg/common"
+	"github.com/certusone/wormhole/node/pkg/p2p"
 	gossipv1 "github.com/certusone/wormhole/node/pkg/proto/gossip/v1"
-	"github.com/libp2p/go-libp2p"
-	dht "github.com/libp2p/go-libp2p-kad-dht"
-	pubsub "github.com/libp2p/go-libp2p-pubsub"
+	"github.com/certusone/wormhole/node/pkg/supervisor"
```

Comment:
Do we really need this supervisor thing? I'm probably wrong, but I remember it being related to that old disconnect bug in the spy.

Reply:
Unfortunately it seems impossible to run the p2p code without supervisor. It gets supervisor stuff from the context on p2p.Run method.